src/System.Private.Uri/src/System/UriHelper.cs

```diff
@@ -114,141 +116,186 @@ internal static class UriHelper
             return true;
         }
 
-        // - forceX characters are always escaped if found
-        // - rsvd character will remain unescaped
-        //
-        // start    - starting offset from input
-        // end      - the exclusive ending offset in input
+        internal static string EscapeString(
+            string stringToEscape, // same name as public API
+            bool checkExistingEscaped, ReadOnlySpan<bool> unreserved, char forceEscape1 = '\0', char forceEscape2 = '\0')
+        {
+            if (stringToEscape is null)
+            {
+                throw new ArgumentNullException(nameof(stringToEscape));
+            }
+            if (stringToEscape.Length == 0)
+            {
+                return string.Empty;
+            }
+
+            // Get the table of characters that do not need to be escaped.
+            Debug.Assert(unreserved.Length == 0x80);
+            ReadOnlySpan<bool> noEscape = stackalloc bool[0];
```

Comment:
Why not `= default;`? Codegen should be the same, but readability is imho better with _default_.

Same at L178.

Reply:
That won't compile. The compiler needs to see the span local initialized to something stack-based in order to then later assign a stackalloc into it.

Reply:
Didn't expect that -- it is already tracked in https://github.com/dotnet/roslyn/issues/25118, https://github.com/dotnet/csharplang/issues/1130.
Thanks!